### PR TITLE
✨ Implement Consent Wizard Step 4 Data Mapper POST (Part 2) (#262)

### PR DESCRIPTION
**Pull request by mistryrn**
## Description of Changes

Updates the PI Participant Prisma schema to include optional Secondary Contact Information fields, adds a migration to update the PI DAS, and updates the Participant Zod types. PR 2 of 3.

### PI DAS
- Updated the `schema.prisma` and added the optional Secondary Contact Information fields to the `Participant` model
- Created a migration to add the aforementioned fields to the PI DAS

### Types
- Updated the `Participant` entity to add the optional Secondary Contact Information
- Refactored the `ConsentRecontact` type to import Secondary Contact Information fields from `Participant`

### Special Instructions
Before running these changes, you will need to run the new migration in `pi-das`:
```zsh
cd apps/pi-das
pnpm prisma:migrate:dev
```
You should also rebuild the `types` package:
```zsh
# from project root
pnpm build:types
```

## PR Readiness Checklist

- [ ] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [ ] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [ ] Added copyrights to new files

### CODE REVIEW

File: apps/pi-das/prisma/schema.prisma [hunk 1]
```diff
@@ -27,28 +27,31 @@ enum Province {
...
+  keycloakId                  String           @unique
+  mailingAddressStreet        String?
+  mailingAddressCity          String?
+  mailingAddressProvince      Province?
+  mailingAddressPostalCode    String?          @db.Char(6)
+  residentialPostalCode       String?          @db.Char(6)
+  assentFormIdentifier        String?
+  secondaryContactFirstName   String?
+  secondaryContactLastName    String?
+  secondaryContactPhoneNumber String?          @db.Char(10)
```
Comment (mistryrn) [inline apps/pi-das/prisma/schema.prisma:54]: Diff shows lots of changes – mostly reformatting – these 3 optional fields are the only real changes.

File: packages/types/src/entities/ConsentRecontact.ts [hunk 2]
```diff
@@ -20,17 +20,10 @@
 import { z } from 'zod';
 
 import { ConsentQuestionId } from './ConsentQuestion.js';
-import { OptionalName, OptionalPhoneNumber } from './fields/index.js';
+import { SecondaryContactInformation } from './Participant.js';
 
 const { RECONTACT__FUTURE_RESEARCH, RECONTACT__SECONDARY_CONTACT } = ConsentQuestionId.enum;
 
-export const SecondaryContactInformation = z.object({
```
Comment (anncatton) [inline packages/types/src/entities/ConsentRecontact.ts:27]: 👍

File: packages/types/src/services/dataMapper/requests/PIUpdateParticipant.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ import { PIParticipantBase } from '../../../entities/index.js';
 // this type can be generalized to take all possible modifiable PI participant fields, not just those in wizard step 2
 // so include fields here that the wizard doesn't deal with
 // do i need to do any refinements on this one?
+// Note: this is also used for Step 4 SecondaryContactInformation
```
Comment (anncatton) [inline packages/types/src/services/dataMapper/requests/PIUpdateParticipant.ts:28]: 🙏 thanks for adding this comment! future Ann appreciates it

File: packages/types/src/entities/Participant.ts [hunk 4]
```diff
@@ -84,13 +91,15 @@ export const ParticipantMailingFields = z.object({
...
-);
+export const PIParticipantBase = ParticipantIdentityBase.merge(ParticipantMailingFields)
+	.merge(
+		z.object({
+			residentialPostalCode: PostalCode,
+			participantOhipMiddleName: OptionalName,
+			hasOhip: z.boolean().optional(),
+		}),
+	)
+	.merge(SecondaryContactInformation);
```
Comment (anncatton) [inline packages/types/src/entities/Participant.ts:102]: 👍 
Just to note, for the UI schema we will probably use a completely separate schema due to all the empty string/undefined issues with the form validation.

**Review by anncatton — APPROVED** [review]
looks good!

**Pull request merged**